File: app/src/main/java/com/example/weizhenbin/wangebug/modules/code/adapters/CodeProjectListAdapter.java
package com.example.weizhenbin.wangebug.modules.code.adapters;

import android.content.Context;
import android.os.Build;
import android.support.annotation.Nullable;
import android.text.Html;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.weizhenbin.wangebug.R;
import com.example.weizhenbin.wangebug.base.WebActivity;
import com.example.weizhenbin.wangebug.image.DefImageConfig;
import com.example.weizhenbin.wangebug.image.ImageLoader;
import com.example.weizhenbin.wangebug.modules.code.entity.ProjectListDataBean;

import java.util.List;

import static android.text.Html.FROM_HTML_MODE_COMPACT;

/**
 * Created by weizhenbin on 2018/8/28.
 */

public class CodeProjectListAdapter extends BaseQuickAdapter<ProjectListDataBean.DataBean.DatasBean,BaseViewHolder> {

    private Context context;

    public CodeProjectListAdapter(final Context context, @Nullable final List<ProjectListDataBean.DataBean.DatasBean> data) {
        super(R.layout.project_list_item,data);
        this.context=context;
        setOnItemChildClickListener(new OnItemChildClickListener() {
            @Override
            public void onItemChildClick(BaseQuickAdapter adapter, View view, int position) {
                if (data!=null){
                    WebActivity.startActivity(context,data.get(position).getLink());
                   // WebActivity.requestPermission(context,"https://github.com/caiyonglong/Notepad/issues");
                }
            }
        });
    }

    @Override
    protected void convert(BaseViewHolder helper, ProjectListDataBean.DataBean.DatasBean item) {
        ImageView imageView=helper.getView(R.id.iv_pic);
        if (TextUtils.isEmpty(item.getEnvelopePic())){
            imageView.setVisibility(View.GONE);
        }else {
            imageView.setVisibility(View.VISIBLE);
            ImageLoader.getImageLoader().imageLoader(context,imageView,item.getEnvelopePic(), DefImageConfig.smallImageLong());
        }
        if (Build.VERSION.SDK_INT>=Build.VERSION_CODES.N){
            helper.setText(R.id.tv_title,Html.fromHtml(item.getTitle(),FROM_HTML_MODE_COMPACT));
            helper.setText(R.id.tv_desc, Html.fromHtml(item.getDesc(),FROM_HTML_MODE_COMPACT));
        }else {
            helper.setText(R.id.tv_title,Html.fromHtml(item.getTitle()));
            helper.setText(R.id.tv_desc, Html.fromHtml(item.getDesc()));
        }
        helper.setText(R.id.tv_chapterName,item.getChapterName());
        helper.setText(R.id.tv_niceDate,item.getNiceDate());
        helper.addOnClickListener(R.id.ll_item);

    }
}

File: app/src/main/java/com/example/weizhenbin/wangebug/modules/code/adapters/CodeProjectListAdapter.kt
package com.example.weizhenbin.wangebug.modules.code.adapters

import android.content.Context
import android.os.Build
import android.text.Html
import android.text.Html.FROM_HTML_MODE_COMPACT
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import com.chad.library.adapter.base.BaseQuickAdapter.OnItemChildClickListener
import com.chad.library.adapter.base.BaseViewHolder
import com.example.weizhenbin.wangebug.R
import com.example.weizhenbin.wangebug.base.BaseSimpleAdapter
import com.example.weizhenbin.wangebug.base.UrlTypeEnum
import com.example.weizhenbin.wangebug.base.WebActivity
import com.example.weizhenbin.wangebug.image.DefImageConfig
import com.example.weizhenbin.wangebug.image.ImageLoader
import com.example.weizhenbin.wangebug.modules.code.entity.ProjectListDataBean

/**
 * Created by weizhenbin on 2018/8/28.
 */

class CodeProjectListAdapter(private val context: Context, data: List<ProjectListDataBean.DataBean.DatasBean>?) : BaseSimpleAdapter<ProjectListDataBean.DataBean.DatasBean, BaseViewHolder>(context, R.layout.project_list_item, data) {

    init {
        onItemChildClickListener = OnItemChildClickListener { adapter, view, position ->
            if (data != null) {
                val datasBean = data[position]
                WebActivity.startActivity(context, datasBean.link!!, datasBean.title, UrlTypeEnum.CODE)
                // WebActivity.requestPermission(context,"https://github.com/caiyonglong/Notepad/issues");
            }
        }
    }

    override fun convert(helper: BaseViewHolder, item: ProjectListDataBean.DataBean.DatasBean) {
        val imageView = helper.getView<ImageView>(R.id.iv_pic)
        if (TextUtils.isEmpty(item.envelopePic)) {
            imageView.visibility = View.GONE
        } else {
            imageView.visibility = View.VISIBLE
            ImageLoader.getImageLoader().imageLoader(context, imageView, item.envelopePic!!, DefImageConfig.smallImageLong())
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            helper.setText(R.id.tv_title, Html.fromHtml(item.title, FROM_HTML_MODE_COMPACT))
            helper.setText(R.id.tv_desc, Html.fromHtml(item.desc, FROM_HTML_MODE_COMPACT))
        } else {
            helper.setText(R.id.tv_title, Html.fromHtml(item.title))
            helper.setText(R.id.tv_desc, Html.fromHtml(item.desc))
        }
        helper.setText(R.id.tv_chapterName, item.chapterName)
        helper.setText(R.id.tv_niceDate, item.niceDate)
        helper.addOnClickListener(R.id.ll_item)

    }
}
